src/lib.rs

```diff
@@ -108,4 +110,23 @@ mod test {
         assert_eq!(a, b);
         assert_eq!(hash(&a), hash(&b));
     }
+
+    #[test]
+    fn test_borrow_impl() {
+        let owned: Cow<'static, str> = Cow::Owned(String::from("hElLO"));
+        let static_borrow: Cow<'static, str> = "Hello".into();
+
+        let mut map = HashMap::new();
+        let val = 42;
+        map.insert(UniCase(owned), val);
+
+        assert_eq!(map.get(&UniCase(static_borrow)), Some(&val));
+
+        fn get_shorter<'map, 'key>(map: &'map HashMap<UniCase<Cow<'static, str>>, u32>, s: &'key str) -> Option<&'map u32>{
+            let fn_borrow: Cow<'key, str> = s.into();
+            map.get::<UniCase<Cow<'key, str>>>(&UniCase(fn_borrow))
```

Comment:
This `get` works with and without the type-hint on stable (1.4), fails in both cases on beta and nightly.